build_runner/lib/src/asset/reader.dart

```diff
@@ -58,6 +60,11 @@ class SinglePhaseReader implements AssetReader {
   }
 
   @override
-  Iterable<AssetId> findAssets(Glob glob) =>
-      _delegate.findAssets(glob).where(_isReadable);
+  Iterable<AssetId> findAssets(Glob glob) => _assetGraph.allNodes
+      .where((n) => n.id.package == _rootPackage)
```

Comment:
Probably worth filing a bug to optimize the graph for filtering by package, not something you need to address right now though.